gobblin-core/src/main/java/gobblin/writer/AvroHdfsDataWriter.java

```diff
@@ -159,15 +119,9 @@ public void commit()
     // the output file if it already exists prevents task retry from being blocked.
     if (this.fs.exists(this.outputFile)) {
       LOG.warn(String.format("Task output file %s already exists", this.outputFile));
-      this.fs.delete(this.outputFile, false);
-    }
-
-    if (!this.fs.rename(this.stagingFile, this.outputFile)) {
-      throw new IOException("Failed to commit data from " + this.stagingFile + " to " + this.outputFile);
+      HadoopUtils.deletePath(this.fs, this.outputFile, false);
     }
-
-    // Setting the same HDFS properties as the original file
-    WriterUtils.setFileAttributesFromState(properties, fs, outputFile);
```

Comment:
I just checked in the code that the preservation of file attributes part has been removed.

I just wanted to confirm whether the removal of setFileAttributesFromState was intentional. The changes were made as part of the feature: Support for having file attributes in the State object #131 (https://github.com/linkedin/gobblin/pull/131) If yes, then WriterUtils.setFileAttributesFromState is no longer required. If no, then this needs to be fixed.